storage/database/src/main/java/de/danoeh/antennapod/storage/database/DBWriter.java

```diff
@@ -521,6 +522,62 @@ private static void removeQueueItemSynchronous(final Context context,
         }
     }
 
+    public static Future<?> moveQueueItemsToTop(final Context context,
+                                            final List<FeedItem> items) {
+        return runOnDbThread(() -> moveQueueItemsSynchronous(context, true, items));
+    }
+
+    public static Future<?> moveQueueItemsToBottom(final Context context,
+                                                   final List<FeedItem> items) {
+        return runOnDbThread(() -> moveQueueItemsSynchronous(context, false, items));
+    }
+
+    private static void moveQueueItemsSynchronous(final Context context,
+                                                  final boolean moveToTop,
+                                                  final List<FeedItem> items) {
+
+        String calledBy = moveToTop ? "moveQueueItemsToTop" : "moveQueueItemsToBottom";
+
+        if (items.isEmpty()) {
+            return;
+        }
+
+        final PodDBAdapter adapter = PodDBAdapter.getInstance();
+        adapter.open();
+        final List<FeedItem> queue = DBReader.getQueue();
+
+        if (moveToTop) {
+            Collections.reverse(items);
+        }
+
+        boolean queueModified = false;
+        List<QueueEvent> events = new ArrayList<>();
+
+        for (FeedItem item : items) {
+            long itemId = item.getId();
+            int index = indexInItemList(queue, itemId);
+            if (index >= 0) {
+                queue.remove(index);
```

Comment:
We don't really need the index, don't we? I mean, we can just assume that items are already in the queue when moving them to the top. If they aren't, nothing bad happens (they just get added). In that case, we can call something like `queue.removeAll(items)` to reduce code complexity a bit